src/Conversion/ONNXToMhlo/Math/Gemm.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===----------------- Gemm.cpp - Lowering Gemm Op ------------------------===//
+//
+// Copyright 2019-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX Gemm Operator to Mhlo dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Support/Debug.h"
+
+#include "src/Conversion/ONNXToMhlo/ONNXToMhloCommon.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+bool closeTo(float a, float b, int ulps = 2) {
+  return std::fabs(a - b) <=
+             std::numeric_limits<float>::epsilon() * std::fabs(a + b) * ulps ||
+         std::fabs(a - b) < std::numeric_limits<float>::min();
+}
+
+template <typename GemmOp>
+struct ONNXGemmOpLoweringToMhlo : public ConversionPattern {
+  ONNXGemmOpLoweringToMhlo(MLIRContext *ctx)
+      : ConversionPattern(GemmOp::getOperationName(), 1, ctx) {}
+
+  void replaceGemmOp(ONNXGemmOp &gemmOp, Operation *op,
+      ONNXGemmOpAdaptor &operandAdaptor, Type elemType,
+      ONNXGemmOpShapeHelper &shapeHelper, ConversionPatternRewriter &rewriter,
+      Location loc) const {
+    float alphaLit = gemmOp.alpha().convertToFloat();
+    float betaLit = gemmOp.beta().convertToFloat();
+    Value A(operandAdaptor.A()), B(operandAdaptor.B()), C(operandAdaptor.C());
+    Value transA = A;
+    Value transB = B;
+    if (gemmOp.transA() == 1) {
+      transA = rewriter.create<mhlo::TransposeOp>(
+          loc, A, rewriter.getI64VectorAttr({1, 0}));
+    }
+    if (gemmOp.transB() == 1) {
+      transB = rewriter.create<mhlo::TransposeOp>(
+          loc, B, rewriter.getI64VectorAttr({1, 0}));
+    }
```

Comment:
Please remove the braces since the body is simple. The same for other `if`. Thanks!